src/runtime/sync.go

```diff
@@ -4,10 +4,14 @@ package runtime
 
 //go:linkname semacquire internal/poll.runtime_Semacquire
 func semacquire(sema *uint32) {
-	panic("todo: semacquire")
+	// TODO the "net" pkg calls this, so panic() isn't an option.  Right
+	// now, just ignore the call.
+	// panic("todo: semacquire")
```

Comment:
Don't do this. There is a very good reason this panics: it is a reminder this should be implemented when it is used.
So either restore the panic, or actually implement `semacquire` and `semrelease`.

(It might be easier to implement the internal/poll package instead in a way that works with TinyGo).

Reply:
Maybe I am missing something here, but since tinygo at the moment only supports a single os thread, it would be sufficient to implement this using a simple mutex, right?

Reply:
No. All this API provides is a `uint32` to store state. That's not enough to store a queue of waiting goroutines and the counter as would be needed for proper semaphore support. I don't think this can even be implemented in the current scheduler - at leat, not with some expensive data structures on the side. It is relatively easy to implement with a futex though, as I have in my threading PR: #4559

Reply:
I see that PR #4559 is still a wip. Maybe we should consider pulling out the futex functionality first to make some progress here?

Reply:
@aykevl does this sound like a good approach to you?